src/components/BrandConfigProvider/styles/StylesProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { merge } from 'lodash';
+import {
+  ComponentStyles,
+  PartialStylesProp,
+  RecursivePartial,
+  StylesBuilder,
+  StylesProp,
+} from '../types';
+import { Theme } from '../theme/Theme';
+import { useTheme } from '../../../hooks';
+
+export const StylesContext = React.createContext({
+  componentStyles: {} as RecursivePartial<ComponentStyles>,
+});
+
+export const useStyles = <T extends StylesBuilder>(
+  name: keyof ComponentStyles,
+  builder: T,
+  ...localStyles: PartialStylesProp<T>[]
+) => {
+  const { theme } = useTheme();
+  const { componentStyles } = React.useContext(StylesContext);
+
+  return React.useMemo(() => {
+    const styles = merge(
```

Comment:
Should we use `StyleSheet.merge` here?

Reply:
Good discussion offline.  Looks like `StyleSheet` doesn't give us anything.